jujugui/static/gui/src/app/utils/changes-utils.js

```diff
@@ -0,0 +1,242 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2012-2013 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('changes-utils', function(Y) {
+
+  var ns = Y.namespace('juju');
+
+  var removeBrackets = /^\(?(.{0,}?)\)?$/;
+
+  /**
+    Util methods for handling ecs changes.
+
+    @method ChangesUtils
+    @constructor
+  */
+  function ChangesUtils(cfg) {}
+
+  ChangesUtils.prototype = {
```

Comment:
These utility methods don't really require this to be an instance. I feel like this should be changed to be a collection of utility methods.

`changesUtils.generateAllDescriptions = function() ...`

Then in places where you reference the other methods you'd instead reference the function directly

`changesUtils.generateChangeDescription( ... )`